endpoints/openrtb2/auction.go

```diff
@@ -1567,3 +1584,30 @@ func getAccountID(pub *openrtb2.Publisher) string {
 	}
 	return metrics.PublisherUnknown
 }
+
+func storedRequestErrorChecker(err error, requestJson []byte, storedRequests map[string]json.RawMessage, storedBidRequestId string) ([]byte, []error) {
+	if err != nil {
+		hasErr, Err := getJsonSyntaxError(requestJson)
+		if hasErr {
+			err = fmt.Errorf("Invalid JSON in Incoming Request: %s", Err)
+		} else {
+			hasErr, Err = getJsonSyntaxError(storedRequests[storedBidRequestId])
+			if hasErr {
+				fmt.Println(err)
+				err = fmt.Errorf("Invalid JSON in Stored Request with ID %s: %s", storedBidRequestId, Err)
+				err = fmt.Errorf("ext.prebid.storedrequest.id refers to Stored Request %s which contains Invalid JSON: %s", storedBidRequestId, Err)
+			}
+		}
+		return nil, []error{err}
+	}
+	return requestJson, nil
+}
+
+func generateUuidForBidRequest() ([]byte, error) {
+	newReqId, err := uuid.NewV4()
+	if err != nil {
+		return nil, err
+	}
+	newReqString := newReqId.String()
+	return []byte(`{"id":"` + newReqString + `"}`), nil
+}
```

Comment:
Similar comment as with the AMP endpoint, could you please extract the random generator to a dependency?